charts/newrelic-infrastructure/templates/daemonset-kubelet.yaml

```diff
@@ -0,0 +1,294 @@
+            - mountPath: /tmp
+              name: tmpfs-tmp
+            {{- with .Values.kubelet.extraVolumeMounts }}
+            {{- tpl . $ | nindent 12 }}
+            {{- end }}
+          {{- with .Values.resources }}
+          resources: {{ toYaml . | nindent 12 }}
+          {{- end }}
+        - name: agent
+          image: {{ .Values.images.agent.repository }}:{{ .Values.images.agent.tag }}
+          imagePullPolicy: {{ .Values.images.agent.pullPolicy }}
+          securityContext:
+            {{- if .Values.privileged }}
+            privileged: true
+            {{- else }}
+            runAsUser: {{ .Values.runAsUser }}
+            runAsGroup: {{ .Values.runAsGroup }}
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: true
+            {{- end }}
+          ports:
+            - containerPort: 8003
+          env:
+            - name: "NRIA_OVERRIDE_HOSTNAME_SHORT"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+            - name: "NRIA_HTTP_SERVER_PORT"
+              value: "8003"
+            - name: NRIA_LICENSE_KEY
+              valueFrom:
+                secretKeyRef:
+                  {{- if (include "newrelic.licenseKey" .) }}
+                  name: {{ template "newrelic.fullname" . }}-config
+                  key: license
+                  {{- else }}
+                  name: {{ include "newrelic.customSecretName" . }}
+                  key: {{ include "newrelic.customSecretLicenseKey" . }}
+                  {{- end }}
+            {{- with .Values.eventQueueDepth }}
+            - name: "NRIA_EVENT_QUEUE_DEPTH"
+              value: {{ .| quote }}
+            {{- end }}
+            {{- /* Ugly hack to check if .Values.enableProcessMetrics is defined: */}}
+            {{- if ne (print .Values.enableProcessMetrics) "<nil>" }}
+            - name: "NRIA_ENABLE_PROCESS_METRICS"
+              value: {{ .Values.enableProcessMetrics | quote }}
+            {{- end }}
+            {{- if (include "newrelic.nrStaging" .) }}
+            - name: "NRIA_STAGING"
+              value: "true"
+            {{- end }}
+            - name: "NRI_KUBERNETES_NODE_NAME"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+            {{- if .Values.useNodeNameAsDisplayName }}
+            - name: "NRIA_DISPLAY_NAME"
+            {{- if .Values.prefixDisplayNameWithCluster }}
+              value: "{{ include "newrelic.cluster" . }}:$(NRI_KUBERNETES_NODE_NAME)"
+            {{- else }}
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+            {{- end }}
+            {{- end }}
+            {{- with .Values.customAttributes }}
+            - name: "NRIA_CUSTOM_ATTRIBUTES"
+              value: {{ . }}
+            {{- end }}
```

Comment:
This should be in all agent containers as well.

Moreover I think the current way of specifying custom attributes, as a json string, is quite awful: We could improve it by letting the user define these as a yaml and then rendering it with `toJson`:

```
# values.yml
customAttributes:
  foo: bar
  team: awesome
```

```suggestion
            {{- with .Values.customAttributes }}
            - name: "NRIA_CUSTOM_ATTRIBUTES"
              value: {{ . | toJson | quote }}
            {{- end }}
```

Reply:
This would break the previous sturcture, but we can create a helper and check if the user defined a string with `kindOf`.